components/p3a/brave_p3a_star_randomness_meta.cc

```diff
@@ -0,0 +1,278 @@
+    return nested_star::get_ppoprf_null_public_key();
+  }
+  return std::move(pk_res.key);
+}
+
+}  // namespace
+
+RandomnessServerInfo::RandomnessServerInfo(
+    uint8_t current_epoch,
+    base::Time next_epoch_time,
+    ::rust::Box<nested_star::PPOPRFPublicKeyWrapper> public_key)
+    : current_epoch(current_epoch),
+      next_epoch_time(next_epoch_time),
+      public_key(std::move(public_key)) {}
+RandomnessServerInfo::~RandomnessServerInfo() {}
+
+BraveP3AStarRandomnessMeta::BraveP3AStarRandomnessMeta(
+    PrefService* local_state,
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
+    RandomnessServerInfoCallback info_callback,
+    BraveP3AConfig* config)
+    : url_loader_factory_(url_loader_factory),
+      local_state_(local_state),
+      info_callback_(info_callback),
+      config_(config) {}
+
+BraveP3AStarRandomnessMeta::~BraveP3AStarRandomnessMeta() {}
+
+void BraveP3AStarRandomnessMeta::RegisterPrefs(PrefRegistrySimple* registry) {
+  registry->RegisterStringPref(kCurrentPKPrefName, std::string());
+  registry->RegisterIntegerPref(kCurrentEpochPrefName, -1);
+  registry->RegisterTimePref(kNextEpochTimePrefName, base::Time());
+}
+
+bool BraveP3AStarRandomnessMeta::VerifyRandomnessCert(
+    network::SimpleURLLoader* url_loader) {
+  if (config_->disable_star_attestation) {
+    VLOG(2) << "BraveP3AStarRandomnessMeta: skipping approved cert check";
+    return true;
+  }
+  const network::mojom::URLResponseHead* response_info =
+      url_loader->ResponseInfo();
+  if (approved_cert_ == nullptr) {
+    LOG(ERROR) << "BraveP3AStarRandomnessMeta: approved cert is missing";
+    AttestServer(false);
+    return false;
+  }
+  if (!response_info->ssl_info.has_value() ||
+      response_info->ssl_info->cert == nullptr) {
+    LOG(ERROR)
+        << "BraveP3AStarRandomnessMeta: ssl info is missing from response info";
+    return false;
+  }
+  if (!response_info->ssl_info->cert->EqualsIncludingChain(
+          approved_cert_.get())) {
+    LOG(ERROR)
+        << "BraveP3AStarRandomnessMeta: approved cert mismatch, will retry "
+           "attestation";
+    AttestServer(false);
+    return false;
+  }
+  return true;
+}
+
+void BraveP3AStarRandomnessMeta::RequestServerInfo() {
+  rnd_server_info_ = nullptr;
+
+  if (!config_->disable_star_attestation && approved_cert_ == nullptr) {
+    AttestServer(true);
+    return;
+  }
+
+  if (!has_used_cached_info) {
+    // Using cached server info, if available.
+    base::Time saved_next_epoch_time =
+        local_state_->GetTime(kNextEpochTimePrefName);
+    if (saved_next_epoch_time > base::Time::Now()) {
+      int saved_epoch = local_state_->GetInteger(kCurrentEpochPrefName);
+      std::string saved_pk = local_state_->GetString(kCurrentPKPrefName);
+
+      rnd_server_info_.reset(new RandomnessServerInfo(
+          static_cast<uint8_t>(saved_epoch), saved_next_epoch_time,
+          DecodeServerPublicKey(&saved_pk)));
```

Comment:
Google's style guides don't specify one way or another, and certainly don't agree with the argument that `new` should be avoided in general, but I tend to agree with my editor's lint that this pattern is more readable as
```c++
rnd_server_info_ = std::make_unique<RandomnessServerInfo>(...);
```
Similar number of characters, and it's more obvious what's going on.

Reply:
yep, i agree. I wrote this when I had trouble using `std::make_unique` in the browser. lately I haven't been having this issue